### PR TITLE
Add COGNITO_USER_POOLS authorizer to API spec

### PR DESCRIPTION
**Pull request by jasonburkert-ccpo**
Added a COGNITO_USER_POOLS type authorizer named `cognitoAuthorizer` to API spec

All API operations now require authorization through `cognitoAuthorizer`

User pool logicalId is now `AtatUserPool`

Ticket: AT-6533

### CODE REVIEW

File: atat_provisioning_wizard_api.yaml [hunk 1]
```diff
@@ -834,6 +834,18 @@ components:
...
+      description: >-
+        Authorizes API access by token. Token obtained from custom IdP and provided in subsequent API requests.
+      type: apiKey
+      name: Authorization
+      in: header
+      x-amazon-apigateway-authtype: cognito_user_pools
+      x-amazon-apigateway-authorizer:
+        type: cognito_user_pools
+        providerARNs:
+          - Fn::GetAtt: ApplicationsApiUserPool.Arn
```
Comment (jasonburkert-ccpo) [inline atat_provisioning_wizard_api.yaml:848]: Need to replace `ApplicationsApiUserPool` here to get the ARN of the pool.

**Comment by kylelaker** [comment]
Good stuff! Should just need a logical ID override!

File: atat_provisioning_wizard_api.yaml (hunk 1 above)
Comment (jasonburkert-ccpo) [reply atat_provisioning_wizard_api.yaml]: Done in 10f27bd5ea564f332396af736d434b72eefcb0a7

**Comment by kylelaker-ccpo** [comment]
Eventually (*not in this ticket*) we should take an action to see if we can make the authorization toggleable.

Something like:

```typescript
const forceAuth = new cdk.CfnCondition(this, "ForceAuthorization", {
  expression: cdk.Fn.conditionEquals(props?.requireAuthorization ?? true, true),
});
forceAuth.overrideLogicalId("IsAuthorizationRequired");
```

and then across the stack:

```yaml
security:
  Fn::If:
    - IsAuthorizationRequired
    - [ { cognitoAuthorizer: [] } ]
    - Ref: AWS::NoValue
```

We're going to have to play around with it to be sure and it feels like maybe it'd be messy to troubleshoot in this PR.

**Comment by kylelaker-ccpo** [comment]
I put the comments above into PR #259. I'd like to see that get merged (into this branch) so that we can handle the configurability option. And this way no one has to make manual changes to the code to disable authorization during their test workflow.

**Review by jeffsegal-ccpo — APPROVED** [review]
Love it

**Comment by jasonburkert-ccpo** [comment]
Good colab.  Thanks everyone!

**Pull request merged**